client/CGameInfo.cpp

```diff
@@ -32,5 +34,14 @@ void CGameInfo::setFromLib()
 	heroh = VLC->heroh;
 	objh = VLC->objh;
 	spellh = VLC->spellh;
+	skillh = VLC->skillh;
 	objtypeh = VLC->objtypeh;
 }
+
+const std::string & CGameInfo::skillInfo(int skill, int level) const
+{
+	const std::string & desc = (*skillh)[SecondarySkill(skill)]->getDescription(level);
+	if(desc.size() > 0)
+		return desc;
+	return generaltexth->skillInfoTexts[skill][level-1];

```

Comment:
generaltexth->skillInfoTexts should be loaded info legacy data of CSkillHandler

Reply:
Is it guaranteed that generaltexth->skillInfoTexts is initialized by the time loadLegacyData is called?

Reply:
1) `CGeneralTextHandler::skillInfoTexts` and `CGeneralTextHandler::skillName` should be removed
2) loading `skillName` and `skillInfoTexts` should be moved to `loadLegacyData` of skill handler
3) if you still want such helper `skillInfo(int skill, int level)`, move if to skill handler
4) if you want to keep skillInfo in CGameInfo make it just redirection to skill handler.

Reply:
Fixed, although there seems to be a bug still with legacy descriptions not being added to json passed to loadFromJson.